build/evergreen_config_lib/variants.py

```diff
@@ -313,6 +317,21 @@ def days(n):
              'test-dns-auth-openssl'
              ],
             {'CC': 'gcc'}),
+    Variant('gcc54',
```

Comment:
Nice, testing compilation and running against one live server on this variant seems like a good idea so we validate the C driver continues to compile against GCC 5.4.